warehouse/query-core/src/main/java/datawave/query/transformer/GroupingTransform.java

```diff
@@ -214,22 +331,30 @@ private void getListKeyCounts(Entry<Key,Document> entry) {
             log.trace("get list key counts for:" + entry);
         }
         keys.add(entry.getKey());
-        int count = 1;
+        
         Set<String> expandedGroupFieldsList = new LinkedHashSet<>();
         // if the incoming Documents have been aggregated on the tserver, they will have a COUNT field.
         // use the value in the COUNT field as a loop max when the fields are put into the multiset
         // During the flush operation, a new COUNT field will be created based on the number of unique
         // field sets in the multiset
-        if (entry.getValue().getDictionary().containsKey("COUNT")) {
-            TypeAttribute countTypeAttribute = ((TypeAttribute) entry.getValue().getDictionary().get("COUNT"));
-            count = ((BigDecimal) countTypeAttribute.getType().getDelegate()).intValue();
+        Map<String,Attribute<? extends Comparable<?>>> dictionary = entry.getValue().getDictionary();
+        Set<String> countKeys = dictionary.keySet().stream().filter(s -> s.startsWith("COUNT")).collect(Collectors.toSet());
```

Comment:
You're making a set here only for the purpose of iterating over it 2 lines down, yes? This could be combined in to something like:
```java
Map<String,Integer> countKeyMap = new HashMap<>();
dictionary.keySet().stream()
    .filter(key -> key.startsWith("COUNT"))
    .filter(countKey -> entry.getValue().getDictionary().containsKey(countKey))
    .forEach(countKey -> {
        TypeAttribute countTypeAttribute = ((TypeAttribute) entry.getValue().getDictionary().get(key));
        int count = ((BigDecimal) countTypeAttribute.getType().getDelegate()).intValue();
        countKeyMap.put(countKey, count);
    });
```